analyzers/tests/SonarAnalyzer.UnitTest/Rules/DatabasePasswordsShouldBeSecureTest.cs

```diff
@@ -57,6 +61,35 @@ public class DatabasePasswordsShouldBeSecureTest
                                     ParseOptionsHelper.FromCSharp8,
                                     GetReferences("3.1.11", "3.19.80"));
 
+        [DataTestMethod]
+        [DataRow(@"TestCases\WebConfig\DatabasePasswordsShouldBeSecure\Values")]
+        [DataRow(@"TestCases\WebConfig\DatabasePasswordsShouldBeSecure\UnexpectedContent")]
+        [TestCategory("Rule")]
+        public void DatabasePasswordsShouldBeSecure_CS_WebConfig(string root)
+        {
+            var webConfigPath = Path.Combine(root, WebConfig);
+            DiagnosticVerifier.VerifyExternalFile(
+                SolutionBuilder.Create().AddProject(AnalyzerLanguage.CSharp).GetCompilation(),
+                new CS.DatabasePasswordsShouldBeSecure(),
+                File.ReadAllText(webConfigPath),
+                TestHelper.CreateSonarProjectConfig(root, TestHelper.CreateFilesToAnalyze(root, webConfigPath)));
+        }
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void DatabasePasswordsShouldBeSecure_CS_CorruptAndNonExistingWebConfigs()
```

Comment:
I think the test name should specify what is the expected behavior for corrupt or missing files.